annotations/openshift-annotations/src/main/java/io/ap4k/openshift/generator/S2iBuildGenerator.java

```diff
@@ -66,11 +66,17 @@ default void on(ConfigurationSupplier<S2iConfig> config) {
   }
 
   default void onClosed() {
-    S2iConfig s2iConfig = session.configurators().get(S2iConfig.class).orElse(DEFAULT_SOURCE_TO_IMAGE_CONFIG);
-    String name = session.configurators().get(OpenshiftConfig.class).map(c -> c.getName()).orElse(getProject().getBuildInfo().getName());
-    if (s2iConfig.isAutoBuildEnabled() || s2iConfig.isAutoDeployEnabled()) {
-      OcBuildHook hook = new OcBuildHook(name, s2iConfig, getProject(), getOutputDirectory());
-      hook.register();
+    ClassLoader tccl = Thread.currentThread().getContextClassLoader();
```

Comment:
Can we have a comment on why changing the tccl is needed here?

Reply:
Sure

Reply:
:+1: